pkg/space/space.go

```diff
@@ -29,6 +32,42 @@ func NewSpace(userSignup *toolchainv1alpha1.UserSignup, targetClusterName string
 	return space
 }
 
+// NewSpaceWithFeatureToggles is the same as NewSpace() but also does a feature toggle lottery drawing
+// and adds the corresponding feature annotations for features which "won" and should be enabled for the space.
+func NewSpaceWithFeatureToggles(userSignup *toolchainv1alpha1.UserSignup, targetClusterName string, compliantUserName, tier string, config toolchainconfig.ToolchainConfig) *toolchainv1alpha1.Space {
+	s := NewSpace(userSignup, targetClusterName, compliantUserName, tier)
+	addFeatureToggles(s, config)
+	return s
+}
+
+// addFeatureToggles does "lottery" drawing for all feature toggles defined in the configuration according to their weights.
+// And it adds the corresponding feature annotation to the space for features that won and should be enabled for the space.
+func addFeatureToggles(space *toolchainv1alpha1.Space, config toolchainconfig.ToolchainConfig) {
+	toggles := config.Tiers().FeatureToggles()
```

Comment:
it seems to me that the only field from the `ToolchainConfig` we are interested in is the list of `FeatureToggle`. I'd propose to pass just it instead of the whole `config`.

```suggestion
func NewSpaceWithFeatureToggles(userSignup *toolchainv1alpha1.UserSignup, targetClusterName string, compliantUserName, tier string, toggles []toolchainconfig.FeatureToggle) *toolchainv1alpha1.Space {
	s := NewSpace(userSignup, targetClusterName, compliantUserName, tier)
	addFeatureToggles(s, toggles)
	return s
}

// addFeatureToggles does "lottery" drawing for all feature toggles defined in the configuration according to their weights.
// And it adds the corresponding feature annotation to the space for features that won and should be enabled for the space.
func addFeatureToggles(space *toolchainv1alpha1.Space, toggles []toolchainconfig.FeatureToggle) {
```

Reply:
https://github.com/codeready-toolchain/host-operator/pull/1048/commits/461d542c5a2a88392e9d0c4caa00e0ad225a1fe4